tfhe/src/core_crypto/gpu/mod.rs

```diff
@@ -90,12 +90,15 @@ impl CudaStream {
         T: Numeric + Into<u64>,
     {
         let dest_size = dest.len() * std::mem::size_of::<T>();
-        cuda_memset_async(
-            dest.as_mut_c_ptr(),
-            value.into(),
-            dest_size as u64,
-            self.as_c_ptr(),
-        );
+        // We have to check that dest is not empty, because cuda_memset with size 0 is invalid
+        if !dest.is_empty() {
```

Comment:
nitpick: we have dest_size already containing the info about copy size so could be dest_size > 0